components/prism-elisp.js

```diff
@@ -0,0 +1,192 @@
+// Functions to construct regular expressions
+// simple form
+// e.g. (interactive ... or (interactive)
+function simple_form(name) {
+	return new RegExp("(\\()" + name + "(?=[\\s\\)])")
+}
+// booleans and numbers
+function primitive(pattern) {
+	return new RegExp("([\\s([])" + pattern + "(?=[\\s)])")
+}
+
+// Patterns in regular expressions
+
+// Symbol name. See https://www.gnu.org/software/emacs/manual/html_node/elisp/Symbol-Type.html
+// & and : are excluded as they are usually used for special purposes
+var symbol = "[-+*/_~!@$%^=<>{}\\w]+"
+// symbol starting with & used in function arguments
+var marker = "&" + symbol
+// Open parenthesis for look-behind
+var par = "(\\()"
+var endpar = "(?=\\))"
+// End the pattern with look-ahead space
+var space = "(?=\\s)"
+
+var language = {
+	// Three or four semicolons are considered a heading.
+	// See https://www.gnu.org/software/emacs/manual/html_node/elisp/Comment-Tips.html
+	heading: {
+		pattern: /;;;.*/,
+		alias: ["comment", "title"],
+	},
+	comment: /;.*/,
+	string: {
+		pattern: /"(?:[^"\\]*|\\.)*"/,
+		greedy: true,
+		inside: {
+			argument: /[-A-Z]+(?=[.,\s])/,
+			symbol: new RegExp("`" + symbol + "'"),
+		},
+	},
+	"quoted-symbol": {
+		pattern: new RegExp("#?'" + symbol),
+		alias: ["variable", "symbol"],
+	},
+	"lisp-property": {
+		pattern: new RegExp(":" + symbol),
+		alias: "property",
+	},
+	splice: {
+		pattern: new RegExp(",@?" + symbol),
+		alias: ["symbol", "variable"],
+	},
+	keyword: [
+		{
+			pattern: new RegExp(
+				par +
+					"(?:(?:lexical-)?let\\*?|(?:cl-)?letf|if|when|while|unless|cons|cl-loop|and|or|not|cond|setq|error|message|null|require|provide|use-package)" +
+					space
+			),
+			lookbehind: true,
+		},
+		{
+			pattern: new RegExp(
+				par + "(?:for|do|collect|return|finally|append|concat|in|by)" + space
+			),
+			lookbehind: true,
+		},
+	],
```

Comment:
Couldn't those two patterns be merged into one?

Reply:
I can try to combine these, but I'm finding it challenging. If it's a required change, I'll definitely work through it, but I'm going to push an update without it in case it isn't